command/snapshot/inspect/snapshot_inspect.go

```diff
@@ -80,6 +103,20 @@ func (c *cmd) Run(args []string) int {
 		return 1
 	}
 
+	// automatically set kvDetails to true if a depth or filter
+	// is provided. this allows the user to omit the -kvdetails
+	// flag if they prefer.
+	if c.kvDepth != 0 || c.kvFilter != "" {
+		c.kvDetails = true
+	}
+
+	// set the default depth if one wasn't specified with -kvdepth.
+	// this is used rather than the flag default to facilitate the
+	// above shortcut.
+	if c.kvDetails && c.kvDepth == 0 {
```

Comment:
Would it be easier to check this inside `kvEnhance`  instead? 
Like around line 310 it looks like on my screen, where we are using this?

Reply:
Moving it there makes sense. I'm just hesitant to do that since it's possible future changes could get added without realizing that's there and that could break expected behavior down the road. Probably not likely, but having it where the flags are initialized would ensure the behavior is consistent with future changes as well.